internal/controllers/clusterextension_controller.go

```diff
@@ -184,11 +213,11 @@ func (r *ClusterExtensionReconciler) handleResolutionErrors(ext *ocv1alpha1.Clus
 	if errors.As(err, &aggErrs) {
 		for _, err := range aggErrs.Errors() {
 			errorMessage := err.Error()
-			if strings.Contains(errorMessage, "no package") {
+			if errors.As(err, &NoPackageError{}) {
```

Comment:
Are you able to verify that this works as expected? IIRC, the last time I did something like this I discovered it did not work as expected and had to do it this way: https://github.com/operator-framework/catalogd/blob/fa4a29bae42a5725d932b1bf6c31381f185e99f7/pkg/controllers/core/clustercatalog_controller.go#L79-L82

Reply:
Semi-off topic, but @everettraven, that recoverable error thing is a feature built directly into controller-runtime since 0.15: https://github.com/kubernetes-sigs/controller-runtime/blob/735b6073bb253c0449bfcf6641855dcf2118bb15/pkg/reconcile/reconcile.go#L105-L109